lib/src/utilities/dart_type_utilities.dart

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.utilities.dart_type_utilities;
+
+import 'package:analyzer/dart/element/type.dart';
+
+class DartTypeUtilities {
+  static List<InterfaceType> findImplementedInterfaces(InterfaceType type,
```

Comment:
Yes, exactly. I was going to comment that I would name it differently, but when I went to see how similar methods are named elsewhere I discovered that we already have what you're looking for. (Sorry I didn't remember this earlier; I should have looked more closely.) Try `type.element.allSupertypes` in place of `findImplementedInterface(type)`.